From f529b8effe692d843141974cb2b3d8ba65d2df24 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 23 Apr 2025 11:35:57 +0800
Subject: [PATCH 15/15] AOSCOS: fix: disable 2D canvas acceleration on Loongson
 LG100/110 graphics

2D-accelerated canvas causes corrupted webpage rendering on Loongson
LG100/110 graphics. Disabling (using software rendering) as a workaround.

Link: https://github.com/AOSC-Dev/chromium-loongarch64/blob/chromium-135.0.7049.114-2/chromium/chromium-135.0.7049.114.4007-loongarch64-gpu.diff
Co-authored-by: Jiajie Chen <c@jia.je>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 gpu/config/software_rendering_list.json | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gpu/config/software_rendering_list.json b/gpu/config/software_rendering_list.json
index 3f09ff3119..611bf3b6d8 100644
--- a/gpu/config/software_rendering_list.json
+++ b/gpu/config/software_rendering_list.json
@@ -1447,6 +1447,18 @@
       "features": [
         "accelerated_2d_canvas"
       ]
+    },
+    {
+      "id": 184,
+      "description": "Corrupted webpage rendering on Loongson LG100/110 graphics with 2D acceleration enabled",
+      "os": {
+        "type" : "linux"
+      },
+      "vendor_id": "0x1014",
+      "device_id": [ "0x7a36" ],
+      "features": [
+        "accelerated_2d_canvas"
+      ]
     }
   ]
 }
-- 
2.49.0

